### PR TITLE
Refactor TiKV auto-scaling

### PR DESCRIPTION
**Pull request by Yisaer**
### What problem does this PR solve? 

1. refactor the tikv auto-scaling controller to be ready for the storage auto-scaling
2. deprecated the de-nosie strategy due to it make auto-scaling in operator too complicated, and we will finally let pd to handle the auto-scaling strategy layer, and auto-scaler controller become the execution layer.

Related changes

 - Need to cherry-pick to the release branch

### Does this PR introduce a user-facing change?:

```release-note
Remove  `readyToScaleThresholdSeconds` in `TidbClusterAutoScaler`, Operator won't support de-noise in `TidbClusterAutoScaler`
```

### CODE REVIEW

**Comment by cofyc** [comment]
Suggested release notes: `Deprecate `readyToScaleThresholdSeconds` in `TidbClusterAutoScaler` as the operator will no longer support de-noise strategy`

File: pkg/apis/pingcap/v1alpha1/tidbclusterautoscaler_types.go [hunk 1]
```diff
@@ -85,6 +85,7 @@ type TidbClusterAutoScalerSpec struct {
 type TikvAutoScalerSpec struct {
 	BasicAutoScalerSpec `json:",inline"`
 
+	// Deprecated
```
Comment (cofyc) [inline pkg/apis/pingcap/v1alpha1/tidbclusterautoscaler_types.go:88]: in which version you plan to remove these fields? you can add a note about this, e.g. `Deprecated in v1.1.3, planned for removal in v1.x.`

File: pkg/apis/pingcap/v1alpha1/tidbclusterautoscaler_types.go [hunk 2]
```diff
@@ -183,6 +184,7 @@ type TikvAutoScalerStatus struct {
 // +k8s:openapi-gen=true
 // BasicAutoScalerStatus describe the basic auto-scaling status
 type BasicAutoScalerStatus struct {
+	// Deprecated
```
Comment (cofyc) [inline pkg/apis/pingcap/v1alpha1/tidbclusterautoscaler_types.go:187]: ditto

File: pkg/apis/pingcap/v1alpha1/tidbclusterautoscaler_types.go [hunk 3]
```diff
@@ -193,6 +195,7 @@ type BasicAutoScalerStatus struct {
 	// RecommendedReplicas describes the calculated replicas in the last auto-scaling reconciliation for the component(tidb/tikv)
 	// +optional
 	RecommendedReplicas int32 `json:"recommendedReplicas,omitempty"`
+	// Deprecated
```
Comment (cofyc) [inline pkg/apis/pingcap/v1alpha1/tidbclusterautoscaler_types.go:198]: ditto

File: pkg/apis/pingcap/v1alpha1/tidbclusterautoscaler_types.go (hunk 1 above)
Comment (Yisaer) [reply pkg/apis/pingcap/v1alpha1/tidbclusterautoscaler_types.go]: updated

File: pkg/apis/pingcap/v1alpha1/tidbclusterautoscaler_types.go [hunk 4]
```diff
@@ -85,6 +85,7 @@ type TidbClusterAutoScalerSpec struct {
 type TikvAutoScalerSpec struct {
 	BasicAutoScalerSpec `json:",inline"`
 
+	// Deprecated in v1.1.3, planned for removal in v1.3
```
Comment (DanielZhangQD) [inline pkg/apis/pingcap/v1alpha1/tidbclusterautoscaler_types.go:88]: Why remove this?
It's not reasonable to scale as soon as the metric exceeds the threshold.
BTW, even if the PD will handle the metrics later, we can remove this when the similar function is implemented in PD. WDYT @cofyc

Comment (cofyc) [reply pkg/apis/pingcap/v1alpha1/tidbclusterautoscaler_types.go:88]: I believe this is in @Yisaer's plan, replaced by PD before v1.3.

Comment (Yisaer) [reply pkg/apis/pingcap/v1alpha1/tidbclusterautoscaler_types.go:88]: I think the current de-noise design in operator is too complicated and haven't been verified whether it is necessary. If we decided let PD make the auto-scaling plan, I think operator could only expose api and do the execution. That will be quite simple and clear, easy to know the status and debug.

**Comment by cofyc** [comment]
please update the release note to reflect the latest change

**Pull request merged**